### DISCUSSION TITLE
Query Builder: Select ignore persist fields ?

### DISCUSSION THREAD

**Discussion by dungkaka** [mikro-orm/mikro-orm#5447] [General]
I have entity Test which have "type" field, you can see "type" have decorator {persist: false}.
```ts
@Entity()
class Test {
  @PrimaryKey()
   id: number;
 
   @Property({ persist: false })
   type: string;

   @Property()
    name: string;
}

let qb = await this.em.qb(Test).select(["id", "type", "name"]).where({}).getSingleResult();
``` 

Query now will ingore "type", and just only select "id" and "name". I happen when I update MikroOrm from 6.0.5 to 6.2.0. I'm wonder there is any new feature or something change then cause break ?

```ts
select "t0"."id", "t0"."name" from "qa"."test" as "t0" limit 1
```

MikroORM: 6.2.0
NodeJs: 20

**Comment by B4nan** [comment]
Why do you mark it as `persist: false` if it's a real property you want to have in the select clause?

You should be probably able to get around this with a (lazy) formula, depends on what you are exactly expecting. `persist: false` means the property is virtual, it cannot be part of a select clause as it does not exist in the table - as opposed to a formula, which is also a virtual property, but backed by an SQL expression (which can resolve to anything, including a real column).

```ts
@Formula(a => `${a}.type`, { lazy: true })
type: string;
```

But then, it's effectively the same as just `lazy: true`, instead of your `persist: false`. Which brings me to my initial question - what are you trying to do?
